x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -2075,4 +2082,17 @@ public void close() {
             }
         }
     }
+
+    private static String getFileExtension(String resourceDescription) {
+        // TODO: consider introspecting resourceDescription for compound files
+        if (resourceDescription.endsWith(LuceneFilesExtensions.CFS.getExtension())) {
+            return LuceneFilesExtensions.CFS.getExtension();
+        }
+        try {
+            LuceneFilesExtensions luceneFilesExtensions = LuceneFilesExtensions.fromFile(resourceDescription);
+            return luceneFilesExtensions != null ? luceneFilesExtensions.getExtension() : NON_LUCENE_EXTENSION_TO_RECORD;
+        } catch (Throwable t) {
```

Comment:
Let's allow Errors to pop up here so we spot critical issues (and allow for process termination) ? Should we catch Exception?

Reply:
> Also, should we open a linked serverless PR?

yes, it's coming

Reply:
ok, now I realize what I had put `Throwable` there in first place: if the `resourceDescription` doesn't have a Lucene file extension, the related `AssertionError` is thrown. This happens in the serverless tests, for example.

Reply:
I'd get rid of that assertion error as opposed to catching it.